ruby/lib/rspec/queue.rb

```diff
@@ -148,6 +149,54 @@ def queue_config
 
     RSpec::Core::Parser.prepend(ParserExtension)
 
+    module ExampleExtension
+      protected
+
+      def mark_as_requeued!(reporter)
+        @exception = nil
+        @metadata = @metadata.dup
```

Comment:
This looks unnecessary to me, I'm probably missing something - could you explain?

Reply:
Right.

So `Example` hold a `@metadata` `Hash`, and delegates someattributes to it: https://github.com/rspec/rspec-core/blob/7dd5edbf664a2213aabf21d06ebe537b8196089d/lib/rspec/core/example.rb#L48-L53

So `Example#execution_result` is actually `Example#@metadata[:execution_result]`. 

Since we want to mutate the result of the cloned example, without modifying to original one, we need to copy `@metadata` as well.

I'll add a comment to make this clearer.